libs/pulumi-static-site/src/static-site.ts

```diff
@@ -0,0 +1,192 @@
+
+        const primaryDomain = args.primaryDomain
+        const redirectDomains = pulumi
+            .all([primaryDomain, args.redirectDomains])
+            .apply(([primaryDomain, redirectDomains]) => {
+                const uniqueRedirectDomains = new Set(redirectDomains ?? [])
+                uniqueRedirectDomains.delete(primaryDomain)
+                return [...uniqueRedirectDomains]
+            })
+
+        const cert = new aws.acm.Certificate(
+            `${name}-cert`,
+            {
+                domainName: primaryDomain,
+                subjectAlternativeNames: redirectDomains,
+                validationMethod: 'DNS',
+            },
+            { parent: this },
+        )
+
+        const zones = pulumi
+            .all([primaryDomain, redirectDomains])
+            .apply(([primaryDomain, redirectDomains]) =>
+                [...new Set([primaryDomain, ...redirectDomains])].map(
+                    (domain) => ({
+                        domain,
+                        zoneId: pulumi.output(
+                            aws.route53
+                                .getZone({ name: domain }, { parent: this })
+                                .catch((e) => {
+                                    throw new pulumi.ResourceError(
+                                        `unable to get zone id for domain ${domain}: ${e}`,
+                                        this,
+                                    )
+                                }),
+                        ).id,
+                    }),
+                ),
+            )
+
+        const acmCertificateArn = zones.apply(
+            (zones) =>
+                new ValidateCertificate(
+                    `${name}-cert`,
+                    {
+                        cert,
+                        zones,
+                    },
+                    { parent: this },
+                ).validCertificateArn,
+        )
+
+        const primaryBucket = new Bucket(
+            `${name}-primary`,
+            {},
+            { parent: this },
+        )
+
+        this.primaryBucket = primaryBucket.bucket
+
+        this.primaryDistribution = new Distribution(
+            `${name}-primary`,
+            {
+                acmCertificateArn,
+                domains: [primaryDomain],
+                originDomainName: this.primaryBucket.websiteEndpoint,
+                priceClass: args.priceClass,
+                cachePolicyId: args.cachePolicyId,
+                originRequestPolicyId: args.originRequestPolicyId,
+            },
+            { parent: this },
+        ).distribution
+
+        primaryBucket.invalidateCloudfrontOnObjectChange({
+            distributionId: this.primaryDistribution.id,
+            distributionArn: this.primaryDistribution.arn,
+        })
+
+        const redirectBucket = new Bucket(
+            `${name}-redirects`,
+            {
+                website: {
```

Comment:
Why do we need a bucket here? Couldn't it just be a behavior on the CDN?

Reply:
Not without using lambda/lambda@edge/cf-functions. It's a tradeoff of execution cost vs bucket cost, and bucket cost is basically zero.